integration-tests/emulated/tests/encointer/encointer-kusama/src/tests/remote_treasury_payout.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) 2023 Encointer Association
+// This file is part of Encointer
+//
+// Encointer is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Encointer is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Encointer.  If not, see <http://www.gnu.org/licenses/>.
+
+use crate::*;
+use emulated_integration_tests_common::{xcm_emulator::ConvertLocation, USDT_ID};
+use encointer_kusama_runtime::{
+	treasuries_xcm_payout::{ConstantKsmFee, GetRemoteFee},
+	xcm_config::KsmLocation,
+	AccountId, CommunityIdentifier,
+};
+use frame_support::{
+	assert_ok,
+	traits::{fungible::Mutate as M, fungibles::Mutate},
+};
+use kusama_system_emulated_network::asset_hub_kusama_emulated_chain::AssetHubKusamaParaPallet;
+use polkadot_runtime_common::impls::VersionedLocatableAsset;
+use xcm::latest::Junctions::X2;
+use xcm_runtime_apis::fees::runtime_decl_for_xcm_payment_api::XcmPaymentApiV1;
+
+fn remote_fee() -> u128 {
+	let fee_asset = ConstantKsmFee::get_remote_fee(Xcm::new(), None);
+	let Asset { id: _, ref fun } = fee_asset;
+
+	match fun {
+		Fungible(fee) => *fee,
+		NonFungible(_) => panic!("Invalid fee"),
+	}
+}
+
+fn treasury_account(maybe_community_identifier: Option<CommunityIdentifier>) -> AccountId {
+	<EncointerKusama as TestExt>::execute_with(|| {
+		encointer_kusama_runtime::EncointerTreasuries::get_community_treasury_account_unchecked(
+			maybe_community_identifier,
+		)
+	})
+}
+
+fn treasury_location_on_ah() -> Location {
+	// Transact the parents native asset on parachain 1000.
+	let asset_kind = VersionedLocatableAsset::V5 {
+		location: (Parent, Parachain(1000)).into(),
+		asset_id: v5::AssetId(Location::parent()),
+	};
+
+	let treasury_account = treasury_account(None);
+
+	<EncointerKusama as TestExt>::execute_with(|| {
+		encointer_kusama_runtime::TransferOverXcm::from_on_remote(
+			&treasury_account,
+			asset_kind.clone(),
+		)
+		.unwrap()
+	})
+}
+
+fn treasury_account_on_ah() -> AccountId {
```

Comment:
```suggestion
fn encointer_treasury_sov_account_on_ah() -> AccountId {
```

Would be less confusing